eth/gasprice/gasprice.go

```diff
@@ -262,7 +263,9 @@ func (gpo *Oracle) getBlockPrices(ctx context.Context, blockNum uint64, limit in
 		sender, _ := tx.GetSender()
 		if err == nil && sender != block.Coinbase() {
 			heap.Push(s, tip)
-			if s.Len() >= limit {
+			count = count + 1
+			// verify if reached max samples(limit) for current block
+			if count >= limit {
```

Comment:
It's terminal condition of loop, then better put inside `for {`, like `for count < limit  && txs.Len() > 0 {`

Reply:
Sw updated